tree_unique_args/src/sugar.egg

```diff
@@ -10,3 +10,13 @@
              (All (Sequential) (Cons a (Cons b (Nil))))
              1)
          :ruleset always-run)
+
+(function NewLoop (i64 Expr Expr) Expr)
+(rewrite (NewLoop id in out)
+         (Loop id (DeepCopyExpr in id) (DeepCopyExpr out id))
+         :ruleset always-run)
+
+(function NewLet (i64 Expr Expr) Expr)
+(rewrite (NewLet id in out)
+         (Let id (DeepCopyExpr in id) (DeepCopyExpr out id))
```

Comment:
Do we need to copy here? A Let doesn't introduce path conditions so maybe not?

Reply:
I just realized we *shouldn't* copy `in`, since it's in the same context, so I updated the rules accordingly. We do still have to copy `out`, since their contained args correspond to the new input), if we want to be able to union args.

(Also, just to be explicit -`NewLet` and `NewBody` are always intended to be called with `(i64-fresh)` as the first argument)

Reply:
Good fix then